data-types/fixed-length-ascii/README.md

```diff
@@ -0,0 +1,33 @@
+# Fixed-length ASCII data type
+
+Defines a data type for fixed-length ASCII strings.
+
+## Permitted fill values
+
+The value of the `fill_value` metadata key must be a string.
```

Comment:
we will need more constraints here, such as as length constraint. for reference, in zarr v2 the fill value was a [base64-encoded ascii string](https://github.com/zarr-developers/zarr-specs/blob/52610ba402fab69e221e4b094213a36c1d658543/docs/v2/v2.0.rst?plain=1#L192-L194) but it didn't constrain the length.